tests/unit/s2n_server_extensions_test.c

```diff
@@ -139,6 +139,45 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_free(conn));
         }
 
+        /* Test TLS13 with Secure Negotiation */
+        {
+            EXPECT_SUCCESS(s2n_enable_tls13());
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            struct s2n_stuffer *hello_stuffer = &conn->handshake.io;
+            conn->secure.server_ecc_params.negotiated_curve = s2n_ecc_supported_curves[0];
+            /* secure renegotiation is requested */
+            conn->secure_renegotiation = 1;
+            /* Test that s2n_server_extensions_send() only works when protocol version is TLS13 */
+            conn->actual_protocol_version = S2N_TLS13;
+
+            /* key_share_send() requires a negotiated_curve */
+            conn->secure.client_ecc_params[0].negotiated_curve = s2n_ecc_supported_curves[0];
+            /* secure_renegotiation extension not send >=TLS13*/
+            uint8_t size = s2n_extensions_server_key_share_send_size(conn)
+                + s2n_extensions_server_supported_versions_size()
+                + EXTENSION_LEN;
+
+            EXPECT_FAILURE(s2n_server_extensions_send(conn, hello_stuffer));
```

Comment:
why is `s2n_server_extensions_send()` failing? can we expect a success here?

Reply:
It succeeds after *s2n_ecc_generate_ephemeral_key*. Check line #167.